### ISSUE TITLE
You may be interested to know about "IFC Normalization"

### ISSUE THREAD

**Issue by highan911** [brunopostle/ifcmerge#11]
Hi Bruno,

Thank you very much for the idea of "Native IFC whitepaper", which is among the references in the paper "A Parallel IFC Normalization Algorithm for Incremental Storage and Version Control". 

The paper can be found at 
    https://arxiv.org/abs/2312.14931

The related project can be found at
    https://github.com/THUCBIMS/CBIMS.IFCNormalization

I believe that you may be interested to know about the project because it provides a different point of view for version comparison and incremental storage of IFC data. 

The motivation for IFC Normalization is slightly different, which is to make IFC comparison also applicable to the IFC files exported from some non-IFC BIM editing tools. I hope the algorithm may help bring some new ideas to the community or to start some new discussions.

I am still working on research about IFC data structure and algorithms. Glad to establish contact with you.

Han Liu

**Comment by brunopostle** [comment]
Thanks Han Liu, this is very interesting. I think this is a good approach to the problem and will be effective for change tracking and long term storage.

Have you tried round-tripping data? Import the normalised IFC into Revit, then normalise IFC exported from this new Revit model - if this result is comparable, then merging and asynchronous editing etc.. is possible.

**Comment by highan911** [comment]
I have not tried importing to Revit; it may be worth trying.

The data loss (especially loss in parametric geometry) in importing IFC to commercial BIM editing tools is a greater problem than exporting, which I think may only be solved by the software companies themselves. Before that, it is risky for modelers to drop the original format and fully rely on IFC.

Anyway, version control on "exporting IFC" and "Native IFC" are much easier : )

**Comment by brunopostle** [comment]
Yes, using just the subset of data that round-trips reliably would limit the available tools (this is the kind of limitation I impose on myself when working with proprietary software)

**Comment by highan911** [comment]
Requirements and limitations in workflow may be in the scope of IDM / IDS. Sounds a good idea
